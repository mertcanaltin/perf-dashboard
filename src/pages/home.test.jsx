import renderer from 'react-test-renderer';

import Home from './home';
import {render, screen} from "@testing-library/react";
test('Home test snapshot', ()=>{
 const homeTest = renderer.create(<Home/>).toJSON();
 expect(homeTest).toMatchSnapshot();
});

test('hello text test',()=>{
 render(<Home/>);
 const helloText = screen.getByText(/Hello Perf-Dashboard/i);
 expect(helloText).toBeInTheDocument();
});

test('Time test', ()=>{
 render(<Home/>);
 const endTime = screen.getByTestId('home-element')
 expect(endTime).toBeInTheDocument();

});






