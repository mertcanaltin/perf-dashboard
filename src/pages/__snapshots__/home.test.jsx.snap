// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home test snapshot 1`] = `
<section
  className="ant-layout home-container"
  data-testid="home-element"
>
  <section
    className="ant-layout"
  >
    <main
      className="ant-layout-content site-layout"
      style={
        Object {
          "marginTop": 64,
          "padding": "0 50px",
        }
      }
    >
      <div
        className="site-layout-background"
        style={
          Object {
            "padding": 10,
          }
        }
      >
        <div
          className="ant-card ant-card-bordered"
        >
          <div
            className="ant-card-body"
            style={Object {}}
          >
            <h4
              className="ant-typography title-hello"
              style={
                Object {
                  "WebkitLineClamp": undefined,
                  "marginTop": 14,
                }
              }
            >
              <strong>
                Hello Perf-Dashboard
              </strong>
            </h4>
            <div
              className="ant-picker"
              onMouseUp={[Function]}
            >
              <div
                className="ant-picker-input"
              >
                <input
                  autoComplete="off"
                  data-testid="timepicker-element"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onMouseDown={[Function]}
                  placeholder="Start Time"
                  readOnly={true}
                  size={10}
                  title="21:04:15"
                  value="21:04:15"
                />
                <span
                  className="ant-picker-suffix"
                >
                  <span
                    aria-label="clock-circle"
                    className="anticon anticon-clock-circle"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="clock-circle"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                      />
                      <path
                        d="M686.7 638.6L544.1 535.5V288c0-4.4-3.6-8-8-8H488c-4.4 0-8 3.6-8 8v275.4c0 2.6 1.2 5 3.3 6.5l165.4 120.6c3.6 2.6 8.6 1.8 11.2-1.7l28.6-39c2.6-3.7 1.8-8.7-1.8-11.2z"
                      />
                    </svg>
                  </span>
                </span>
                <span
                  className="ant-picker-clear"
                  onMouseDown={[Function]}
                  onMouseUp={[Function]}
                  role="button"
                >
                  <span
                    aria-label="close-circle"
                    className="anticon anticon-close-circle"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="close-circle"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
                      />
                    </svg>
                  </span>
                </span>
              </div>
            </div>
            <div
              className="ant-picker"
              onMouseUp={[Function]}
            >
              <div
                className="ant-picker-input"
              >
                <input
                  autoComplete="off"
                  data-testid="timepicker-element"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onMouseDown={[Function]}
                  placeholder="End Time"
                  readOnly={true}
                  size={10}
                  title="21:34:15"
                  value="21:34:15"
                />
                <span
                  className="ant-picker-suffix"
                >
                  <span
                    aria-label="clock-circle"
                    className="anticon anticon-clock-circle"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="clock-circle"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                      />
                      <path
                        d="M686.7 638.6L544.1 535.5V288c0-4.4-3.6-8-8-8H488c-4.4 0-8 3.6-8 8v275.4c0 2.6 1.2 5 3.3 6.5l165.4 120.6c3.6 2.6 8.6 1.8 11.2-1.7l28.6-39c2.6-3.7 1.8-8.7-1.8-11.2z"
                      />
                    </svg>
                  </span>
                </span>
                <span
                  className="ant-picker-clear"
                  onMouseDown={[Function]}
                  onMouseUp={[Function]}
                  role="button"
                >
                  <span
                    aria-label="close-circle"
                    className="anticon anticon-close-circle"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="close-circle"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
                      />
                    </svg>
                  </span>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </main>
  </section>
  <main
    className="ant-layout-content site-layout"
    style={
      Object {
        "marginTop": 64,
        "padding": "0 50px",
      }
    }
  >
    <div
      className="site-layout-background"
      style={
        Object {
          "minHeight": 380,
          "padding": 24,
        }
      }
    >
      <div
        className="ant-row"
        style={
          Object {
            "marginBottom": 28,
            "marginLeft": -12,
            "marginRight": -12,
            "marginTop": -8,
          }
        }
      >
        <div
          className="ant-col ant-col-12"
          style={
            Object {
              "paddingBottom": 8,
              "paddingLeft": 12,
              "paddingRight": 12,
              "paddingTop": 8,
            }
          }
        >
          <div
            className="site-card-wrapper"
          >
            <div
              className="ant-card ant-card-bordered title"
            >
              <div
                className="ant-card-head"
                style={Object {}}
              >
                <div
                  className="ant-card-head-wrapper"
                >
                  <div
                    className="ant-card-head-title"
                  >
                    TTFB
                  </div>
                </div>
              </div>
              <div
                className="ant-card-body"
                style={Object {}}
              >
                <div
                  className="recharts-responsive-container"
                  style={
                    Object {
                      "height": "100%",
                      "maxHeight": undefined,
                      "minHeight": undefined,
                      "minWidth": undefined,
                      "width": "99%",
                    }
                  }
                />
              </div>
            </div>
             
          </div>
        </div>
        <div
          className="ant-col ant-col-12"
          style={
            Object {
              "paddingBottom": 8,
              "paddingLeft": 12,
              "paddingRight": 12,
              "paddingTop": 8,
            }
          }
        >
          <div
            className="site-card-wrapper"
          >
            <div
              className="ant-card ant-card-bordered title"
            >
              <div
                className="ant-card-head"
                style={Object {}}
              >
                <div
                  className="ant-card-head-wrapper"
                >
                  <div
                    className="ant-card-head-title"
                  >
                    FCP
                  </div>
                </div>
              </div>
              <div
                className="ant-card-body"
                style={Object {}}
              >
                <div
                  className="recharts-responsive-container"
                  style={
                    Object {
                      "height": "100%",
                      "maxHeight": undefined,
                      "minHeight": undefined,
                      "minWidth": undefined,
                      "width": "99%",
                    }
                  }
                />
              </div>
            </div>
             
          </div>
        </div>
        <div
          className="ant-col ant-col-12"
          style={
            Object {
              "paddingBottom": 8,
              "paddingLeft": 12,
              "paddingRight": 12,
              "paddingTop": 8,
            }
          }
        >
          <div
            className="site-card-wrapper"
          >
            <div
              className="ant-card ant-card-bordered title"
            >
              <div
                className="ant-card-head"
                style={Object {}}
              >
                <div
                  className="ant-card-head-wrapper"
                >
                  <div
                    className="ant-card-head-title"
                  >
                    DomLoad
                  </div>
                </div>
              </div>
              <div
                className="ant-card-body"
                style={Object {}}
              >
                <div
                  className="recharts-responsive-container"
                  style={
                    Object {
                      "height": "100%",
                      "maxHeight": undefined,
                      "minHeight": undefined,
                      "minWidth": undefined,
                      "width": "99%",
                    }
                  }
                />
              </div>
            </div>
             
          </div>
        </div>
        <div
          className="ant-col ant-col-12"
          style={
            Object {
              "paddingBottom": 8,
              "paddingLeft": 12,
              "paddingRight": 12,
              "paddingTop": 8,
            }
          }
        >
          <div
            className="site-card-wrapper"
          >
            <div
              className="ant-card ant-card-bordered title"
            >
              <div
                className="ant-card-head"
                style={Object {}}
              >
                <div
                  className="ant-card-head-wrapper"
                >
                  <div
                    className="ant-card-head-title"
                  >
                    Window Load
                  </div>
                </div>
              </div>
              <div
                className="ant-card-body"
                style={Object {}}
              >
                <div
                  className="recharts-responsive-container"
                  style={
                    Object {
                      "height": "100%",
                      "maxHeight": undefined,
                      "minHeight": undefined,
                      "minWidth": undefined,
                      "width": "99%",
                    }
                  }
                />
              </div>
            </div>
             
          </div>
        </div>
      </div>
      <table
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <thead
          className="ant-table-thead"
        >
          <tr>
            <th
              className="ant-table-cell url"
            >
              Url
            </th>
            <th
              className="ant-table-cell type"
            >
              Initiator Type
            </th>
            <th
              className="ant-table-cell size"
            >
              Transfer Size
            </th>
            <th
              className="ant-table-cell duration"
            >
              Duration
            </th>
          </tr>
        </thead>
        <tbody
          className="ant-table-tbody"
        />
      </table>
    </div>
  </main>
  <footer
    className="ant-layout-footer"
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    Mert Can Altın ©2021
  </footer>
</section>
`;
